test/storage/sql_table_test.cpp

```diff
@@ -570,10 +589,10 @@ TEST_F(SqlTableTests, ScanTest) {
   uint32_t datname = *reinterpret_cast<uint32_t *>(value);
   EXPECT_EQ(datname, datname_map[id]);
   value = row1.AccessWithNullCheck(pc_pair.second.at(catalog::col_oid_t(2)));
+  uint32_t new_col = *reinterpret_cast<uint32_t *>(value);
```

Comment:
Based off default value semantics, I would expect the following mapping from `ID` to `new_col`:
```
  100 -> new_col_default_value
  200 -> new_col_default_value
  300 -> nullptr
  400 -> new_col_map[id]
```

Specifically, the assumption would be that the `NULL` in 300 is because the query explicitly overrode the default value with `NULL` and the query planner somehow optimized away the insert value.

Reply:
Forcing the value to be `NULL` despite the presence of a default value would be problematic right? How do we know now about which columns were actually `NULL` and which columns were set to `NULL` by the `select`/`scan` functions? For `select` I can see passing in a vector or something which tells the columns it had filled in. But for `scan` we need to maintain the columns filled in for each of the tuples to do this.